compiler/src/iree/compiler/Codegen/Common/test/tile_and_distribute_workgroups_using_forall.mlir

```diff
@@ -0,0 +1,298 @@
+      ins(%arg0 : tensor<f32>) outs(%0 : tensor<f32>)
+      attrs = {lowering_config = #iree_codegen.lowering_config<tile_sizes=[[]]>} {
+    ^bb0(%b0 : f32, %b1 : f32) :
+      %1 = arith.addf %b0, %b0 : f32
+      linalg.yield %1 : f32
+  } -> tensor<f32>
+  return %1 : tensor<f32>
+}
+// CHECK-LABEL: func @scalar(
+//   CHECK-NOT:   scf.forall
+
+// -----
+
+func.func @matmul_interchange(%0 : tensor<?x?xf32>,
+    %1 : tensor<?x?xf32>, %2 : tensor<?x?xf32>) -> tensor<?x?xf32> {
+  %3 = linalg.matmul
+      {lowering_config = #iree_codegen.lowering_config<tile_sizes = [{sizes = [32, 64, 0], interchange = [1, 0, 2]}]>}
+      ins(%0, %1 : tensor<?x?xf32>, tensor<?x?xf32>)
+      outs(%2 : tensor<?x?xf32>) -> tensor<?x?xf32>
+  return %3 : tensor<?x?xf32>
+}
+// CHECK-LABEL: func @matmul_interchange(
+//  CHECK-SAME:     %[[LHS:[a-zA-Z0-9]+]]: tensor<?x?xf32>,
+//  CHECK-SAME:     %[[RHS:[a-zA-Z0-9]+]]: tensor<?x?xf32>,
+//  CHECK-SAME:     %[[INIT:[a-zA-Z0-9]+]]: tensor<?x?xf32>)
+//   CHECK-DAG:   %[[C0:.+]] = arith.constant 0
+//   CHECK-DAG:   %[[C1:.+]] = arith.constant 1
+//   CHECK-DAG:   %[[M:.+]] = tensor.dim %[[LHS]], %[[C0]]
+//   CHECK-DAG:   %[[N:.+]] = tensor.dim %[[RHS]], %[[C1]]
+//       CHECK:   %[[RESULT:.+]] = scf.forall (%[[IV0:[a-zA-Z0-9]+]], %[[IV1:[a-zA-Z0-9]+]])
+//  CHECK-SAME:       to (%[[N]], %[[M]]) step (64, 32)
+//  CHECK-SAME:       shared_outs(%[[OUTS:.+]] = %[[INIT]])
+//   CHECK-DAG:     %[[LHS_SLICE:.+]] = tensor.extract_slice %[[LHS]][%[[IV1]], 0]
+//   CHECK-DAG:     %[[RHS_SLICE:.+]] = tensor.extract_slice %[[RHS]][0, %[[IV0]]
+//   CHECK-DAG:     %[[INIT_SLICE:.+]] = tensor.extract_slice %[[OUTS]][%[[IV1]], %[[IV0]]{{\]}}
+//       CHECK:     %[[MATMUL_SLICE:.+]] = linalg.matmul
+//  CHECK-SAME:         ins(%[[LHS_SLICE]], %[[RHS_SLICE]] :
+//  CHECK-SAME:         outs(%[[INIT_SLICE]] :
+//       CHECK:     scf.forall.in_parallel
+//       CHECK:       tensor.parallel_insert_slice %[[MATMUL_SLICE]] into %[[OUTS]][%[[IV1]], %[[IV0]]]
+//       CHECK:     mapping = [#iree_codegen.workgroup_mapping<y>, #iree_codegen.workgroup_mapping<x>]
+//       CHECK:   return %[[RESULT]]
+
+// -----
+
+func.func @no_compute(%arg0 : memref<?x?x?xf32>, %arg1 : memref<?x?x?xf32>) {
+  memref.assume_alignment %arg0, 64 : memref<?x?x?xf32>
+  memref.assume_alignment %arg1, 64 : memref<?x?x?xf32>
+  return
+}
+// CHECK-LABEL: @no_compute(
+//   CHECK-NOT:   scf.forall
+
+// -----
+
+func.func @matmul_memrefs(%0 : memref<?x?xf32>, %1 : memref<?x?xf32>, %2 : memref<?x?xf32>) {
+  linalg.matmul {lowering_config = #iree_codegen.lowering_config<tile_sizes = [[64, 64, 0]]>}
+      ins(%0, %1 : memref<?x?xf32>, memref<?x?xf32>)
+      outs(%2 : memref<?x?xf32>)
+  return
+}
+// CHECK-LABEL: func @matmul_memrefs(
+//       CHECK:   scf.forall
+//       CHECK:     linalg.matmul
+//       CHECK:   mapping = [#iree_codegen.workgroup_mapping<y>, #iree_codegen.workgroup_mapping<x>]
```

Comment:
can you add (or change) one of the tests to use `iree_gpu.lowering_config<{workgroup = [...]}>`?

Reply:
Ah... that didnt work, but I am not entirely sure why. I think the issue is with `iree_gpu.lowering_config`. Ill triage that as a follow up.

Reply:
Ok darn, following up sounds good then.